### PR TITLE
(First draft) Playground editor

### PR DESCRIPTION
**Pull request by gksander**
This PR just sets up foundation for groqd playground editor. The idea with this is:

- Monaco is the superior in-browser editor and can handle TS out of the box.
- Bundling Monaco is a _pain_, so I'm proposing we eat that cost by hosting the editor from this repo.
- Playgrounds can then just consume the editor via iframe and communicate via `postMessage`.

This allows us to ship a sanity plugin for groqd playground without forcing the user into a complicated build setup involving monaco.

image

### CODE REVIEW

File: packages/playground-editor/src/messaging.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+let targetUrl!: string;
```
Comment (carloskelly13) [inline packages/playground-editor/src/messaging.ts:1]: Do we need to non-null assert this? We're only calling the function which is guaranteed to return `string`.

File: packages/playground-editor/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    };
+  }, []);
+
+  return (
+    <div ref={containerRef} style={{ width: "100vw", height: "100vh" }}></div>
+  );
+}
+
+// Initial code, will likely change in the future.
+const INIT_VALUE = [
```
Comment (carloskelly13) [inline packages/playground-editor/src/App.tsx:105]: I think this is fine for now, but I wonder if consuming it from a template file might be better and easier to update?

**Review by carloskelly13 — APPROVED** [review]
This is pretty neat. I kicked around the tires a bit to see the integrations and I'm impressed it handles auto complete and imports nicely.

Screenshot 2023-04-12 at 4 10 46 PM
Screenshot 2023-04-12 at 4 10 32 PM

File: packages/playground-editor/src/App.tsx (hunk 2 above)
Comment (gksander) [reply packages/playground-editor/src/App.tsx:105]: yeah agreed. At some point, we'll also allow passing in initial value via query param – so this is going to change in the near future.

**Pull request merged**